src/components/UnitCard.js

```diff
@@ -12,27 +16,35 @@ const UpgradeBar = styled.div`
   display: flex;
 `;
 
-const RemoveButton = styled.button`
+const RemoveButton = styled.div`
   float: right;
 `;
 
-const UnitCard = ({ unit, deleteFunction, index }) => {
+const UnitCard = ({ unit, deleteFunction, index, updateFunction }) => {
   const { name, upgrades_slots } = unit;
   return (
     <Body>
       <p>
         {name}
-        <RemoveButton
-          onClick={() => {
-            deleteFunction(index);
-          }}
-        >
-          X{" "}
+        <RemoveButton>
+          <IconButton
+            onClick={() => {
+              deleteFunction(index);
+            }}
+          >
+            <ClearIcon />
+          </IconButton>
         </RemoveButton>
       </p>
       <UpgradeBar>
-        {upgrades_slots.map(name => (
-          <UpgradeButton type={name} />
+        {upgrades_slots.map((name, key) => (
+          <UpgradeButton
+            unitKey={index}
```

Comment:
`key` prop - lint should be reminding you about this one

Reply:
lint wasn't, but console definitely does 😅